xdsl/dialects/func.py

```diff
@@ -177,17 +177,11 @@ def print(self, printer: Printer):
                 printer.print(" ")
         else:
             printer.print_attribute(self.function_type)
-        attr_dict = {
-            k: v
-            for k, v in self.attributes.items()
-            if k not in ("sym_name", "function_type", "sym_visibility")
-        }
-        if len(attr_dict) > 0:
-            printer.print(" attributes {")
-            printer.print_list(
-                attr_dict.items(), lambda i: printer.print(f'"{i[0]}" = {i[1]}')
-            )
-            printer.print("}")
+
+        reserved = ("sym_name", "function_type", "sym_visibility")
```

Comment:
create this outside of function, and make a `set`? Seems like good practice even if it won't make a big difference here probably

Reply:
This is a nit

Reply:
I like the idea, making it more declarative, but then it sounds to me like something that could be more integrated to PyRDL?
I'll do the small change, just giving this idea

Reply:
Here I guess it's less about it being declarative and more about Python not being good at constant value optimisation, I think it'll create a tuple every time which seems like a waste if it's constant

Reply:
We could have it integrated with `PyRDL`, but we would need to slightly change the parser.
For instance, you could already do `reserved = self.irdl_definition.attributes.keys()`, but that would not be future-proof, as we don't actually check that all attributes are printed, so if you add a new attribute, we might miss it later on.
The declarative framework will ensure that though.